### PR TITLE
[TE2] Add a option to set the size of payload in CHIP request message

### LINKED ISSUE #5551: [TE1] Add a option to set the size of payload in request payload
#### Problem
[TestCase: 4.13.4.2-2] Attempts to send a message larger than the agreed upon max message size results in a CHIP_ERROR_MESSAGE_TOO_LONG error and the connection is aborted.

We need a option to set the size of payload to test this test case.

 #### Proposed Solution
Add a option to set the size of payload in request payload

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
 https://docs.google.com/spreadsheets/d/194sxlYKYBoELgWC31vNRF3suY2KSpQVLU5f1cgCEX9g/edit?ts=603edbf6#gid=0
 
[TestCase: 4.13.4.2-2] Attempts to send a message larger than the agreed upon max message size results in a CHIP_ERROR_MESSAGE_TOO_LONG error and the connection is aborted.

We need a option to set the size of payload to test this test case.

 #### Summary of Changes
Add a option to set the size of payload in CHIP request message

 Fixes #5551

### CODE REVIEW

File: examples/shell/shell_common/cmd_ping.cpp [hunk 1]
```diff
@@ -137,10 +144,11 @@ CHIP_ERROR SendEchoRequest(streamer_t * stream)
     CHIP_ERROR err = CHIP_NO_ERROR;
 
     Messaging::SendFlags sendFlags;
-    const char kRequestFormat[] = "Echo Message %" PRIu64 "\n";
+    uint32_t size                   = gPingArguments.GetEchoReqSize();
+    const char kRequestFormat[size] = "Echo Message %" PRIu64 "\n";
```
Comment (LuDuda) [inline examples/shell/shell_common/cmd_ping.cpp:148]: This seems risky to me, as it allocates an array on the stack that may have an extremely large or small size (e.g. smaller than assigned text).

**Review by LuDuda — COMMENTED** [review]
FYI, this feature is not required for TE1, as CRMP was deferred.

File: examples/shell/shell_common/cmd_ping.cpp (hunk 1 above)
Comment (yufengwangca) [reply examples/shell/shell_common/cmd_ping.cpp]: Acked, we should allocate the buffer from heap instead of stack.

**Comment by yufengwangca** [comment]
> FYI, this feature is not required for TE1, as CRMP was deferred.

Oops, just have a meeting today with Chaitanya to review CRMP test cases for TE1

**Review by LuDuda — APPROVED** [review]
Thanks for applying fixes. I'm quite sure that CRMP is not part of TE1, as already a few participants finished it and there were no CRMP test cases defined.

File: examples/shell/shell_common/cmd_ping.cpp [hunk 2]
```diff
@@ -181,7 +192,14 @@ CHIP_ERROR SendEchoRequest(streamer_t * stream)
         gPingArguments.SetWaitingForEchoResp(true);
         gPingArguments.IncrementEchoCount();
     }
-    else
+
+exit:
+    if (requestData != nullptr)
+    {
+        chip::Platform::MemoryFree(requestData);
```
Comment (andy31415) [inline examples/shell/shell_common/cmd_ping.cpp:199]: could we use a RAII buffer here?

We have some scopedbuffer class however I believe that one does not support arrays.

**Review by andy31415 — APPROVED** [review]

File: examples/shell/shell_common/cmd_ping.cpp (hunk 2 above)
Comment (yufengwangca) [reply examples/shell/shell_common/cmd_ping.cpp:199]: Yes, we have scoped buffer class such as PacketBufferHandle for PacketBuffer, but not for array with dynamic size yet.

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/shell/shell_common/cmd_ping.cpp (hunk 2 above)
Comment (andy31415) [reply examples/shell/shell_common/cmd_ping.cpp:199]: I think we should start adding RAII classes as part of tech debt work. Will see if we can add them during our standup.

**Pull request merged**